### PR TITLE
feat: slashed balance accounting

### PR DESCRIPTION
**Pull request by 0xShaito**
# 🤖 Linear

Closes GRT-209

### CODE REVIEW

**Review by ashitakah — DISMISSED** [review]

File: src/contracts/HorizonAccountingExtension.sol [hunk 1]
```diff
@@ -223,16 +226,18 @@ contract HorizonAccountingExtension is Validator, IHorizonAccountingExtension {
...
+
       // Check the balance in the contract
       // If not enough balance, slash some users to get enough balance
-      uint256 _balance = GRT.balanceOf(address(this));
-
       // TODO: How many iterations should we do?
-      while (_balance < _claimAmount) {
-        _balance += _slash(_disputeId, 1, MAX_USERS_TO_CHECK, _result, _status);
+      while (disputeBalance[_disputeId] < _claimAmount) {
+        _slash(_disputeId, 1, MAX_USERS_TO_CHECK, _result, _status);
```
Comment (0xJabberwock) [inline src/contracts/HorizonAccountingExtension.sol:235]: Should be `while (disputeBalance[_disputeId] < _rewardAmount)`.

File: src/contracts/HorizonAccountingExtension.sol [hunk 2]
```diff
@@ -54,6 +54,9 @@ contract HorizonAccountingExtension is Validator, IHorizonAccountingExtension {
   /// @inheritdoc IHorizonAccountingExtension
   mapping(address _caller => bool _authorized) public authorizedCallers;
 
+  /// @inheritdoc IHorizonAccountingExtension
+  mapping(bytes32 _disputeId => uint256 _balance) public disputeBalance;
```
Comment (0xJabberwock) [inline src/contracts/HorizonAccountingExtension.sol:58]: I suggest adding `disputeBalance` assertions to integration tests that call `HorizonAccountingExtension::claimEscalationReward`.

**Review by 0xJabberwock — CHANGES_REQUESTED** [review]

File: src/contracts/HorizonAccountingExtension.sol (hunk 1 above)
Comment (0xShaito) [reply src/contracts/HorizonAccountingExtension.sol:235]: Yes. But this change is in your PR. Didn't want to introduce new changes unrelated to this one

File: src/contracts/HorizonAccountingExtension.sol (hunk 2 above)
Comment (0xShaito) [reply src/contracts/HorizonAccountingExtension.sol:58]: Yes, will be added but now we only have the bondEscalation

Comment (0xShaito) [reply src/contracts/HorizonAccountingExtension.sol:58]: Created a task for it! https://linear.app/defi-wonderland/issue/GRT-213/integration-testing-for-disputebalance-in-horizonaccountingextension

**Review by 0xJabberwock — DISMISSED** [review]

File: src/interfaces/IHorizonAccountingExtension.sol [hunk 3]
```diff
@@ -302,14 +302,6 @@ interface IHorizonAccountingExtension is IValidator {
    */
   function approvedModules(address _user) external view returns (address[] memory _approvedModules);
 
-  /**
-   * @notice Checks whether an address is an authorized caller.
-   *
-   * @param _caller      The address to check
-   * @return _authorized True if the address is authorized, false otherwise
-   */
-  function authorizedCallers(address _caller) external returns (bool _authorized);
```
Comment (0xJabberwock) [inline src/interfaces/IHorizonAccountingExtension.sol:296]: Why remove this from the interface?

Comment (0xShaito) [reply src/interfaces/IHorizonAccountingExtension.sol:296]: It was duplicated in the interface

**Review by 0xJabberwock — APPROVED** [review]

**Review by ashitakah — APPROVED** [review]

**Pull request merged**